sched/scheduler/cluster_state.go

```diff
@@ -302,10 +303,14 @@ func (c *clusterState) update(updates []cluster.NodeUpdate) {
 			}
 		} else if ns.timeFlaky != nilTime && now.Sub(ns.timeFlaky) > c.maxFlakyDuration {
 			// This flaky node has been suspended long enough, try adding it back to the healthy node pool.
-			delete(c.suspendedNodes, ns.node.Id())
-			c.nodes[ns.node.Id()] = ns
+			// TimeFlaky should've been the only time* value set at this point, reset it.
```

Comment:
So we don't have to delete from c.suspendedNodes any more? Or add back into c.nodes?  And no idea why startReadyLoop has to be added.